tests/test_packages/test_protocols/test_fipa.py

```diff
@@ -303,3 +293,132 @@ def test_dialogues():
     dialogues.dialogue_stats.add_dialogue_endstate(FIPADialogue.EndState.DECLINED_CFP, is_self_initiated=False)
     assert dialogues.dialogue_stats.self_initiated == {FIPADialogue.EndState.SUCCESSFUL: 1, FIPADialogue.EndState.DECLINED_PROPOSE: 0, FIPADialogue.EndState.DECLINED_ACCEPT: 0, FIPADialogue.EndState.DECLINED_CFP: 0}
     assert dialogues.dialogue_stats.other_initiated == {FIPADialogue.EndState.SUCCESSFUL: 0, FIPADialogue.EndState.DECLINED_PROPOSE: 0, FIPADialogue.EndState.DECLINED_ACCEPT: 0, FIPADialogue.EndState.DECLINED_CFP: 1}
+    assert dialogues.dialogues_as_seller is not None
+    msg = FIPAMessage(message_id=0,
+                      dialogue_reference=(str(0), ''),
+                      target=1,
+                      performative=FIPAMessage.Performative.ACCEPT)
+    msg.counterparty = "opponent"
+    result = dialogues.is_permitted_for_new_dialogue(msg)
+    assert result is False
+
+
+def test_dialogues_is_belonging_to_registered_dialogue():
+    """Test if the dialogue is a part of an other dialogue."""
+    dialogues = FIPADialogues()
+    result = dialogues.create_opponent_initiated(dialogue_opponent_addr="opponent", dialogue_reference=(str(0), ''), is_seller=True)
+
+    proposal = [
+        Description({"foo1": 1, "bar1": 2}),
+        Description({"foo2": 1, "bar2": 2}),
+    ]
+    last_msg = FIPAMessage(message_id=0,
+                           dialogue_reference=(str(0), ''),
+                           target=0,
+                           performative=FIPAMessage.Performative.PROPOSE,
+                           proposal=proposal)
+    result._outgoing_messages.append(last_msg)
+
+    other_initiated_dialogue = DialogueLabel((str(0), ''), "opponent", "opponent")
+
+    dialogues.dialogues[other_initiated_dialogue] = cast(FIPADialogue, result)
+    msg = FIPAMessage(message_id=1,
+                      dialogue_reference=(str(0), ''),
+                      target=1,
+                      performative=FIPAMessage.Performative.ACCEPT_W_INFORM,
+                      info={"address": "dummy_address"})
+    msg.counterparty = "opponent"
+    r = dialogues.is_belonging_to_registered_dialogue(msg, agent_addr="opponent")
+    assert r
+
```

Comment:
This test needs to be broken up into several parts. Maybe have a test class for initialisation and then test several cases in individual methods